lib/galaxy/datatypes/data.py

```diff
@@ -864,6 +884,23 @@ def regex_line_dataprovider( self, dataset, **settings ):
         dataset_source = dataproviders.dataset.DatasetDataProvider( dataset )
         return dataproviders.line.RegexLineDataProvider( dataset_source, **settings )
 
+    def __safe_readlines(self, path, num_lines=1, size=5 * 1024):
+        """
+        Read lines from the start of a file but be sure to restrict the size of
+        memory that will be consumed and use proper file handling to ensure file
+        handles aren't leaked.
+        """
+        with open(path, "r") as f:
+            start_of_file = f.read(size)
+            lines = start_of_file.splitlines()
+            return lines[0:num_lines]
+
+    def safe_readlines(self, path, num_lines=1, size=5 * 1024):
+        return "".join( self.__safe_readlines( path, num_lines, size ) )
+
+    def safe_readline(self, path):
+        return self.__safe_readlines(path)[0]
+
```

Comment:
Sorry I missed your message, I will fix this tomorrow and use the file.readline function with the size parameter.